### PR TITLE
generic-zuul-obj: get zuul object from yaml

### PR DESCRIPTION
**Pull request by pmpbaptista**
Return zuul objects from yaml based on object type
Include Enum for each object
Tests for each object added

Issue: https://github.com/codesquadnest/zuulcilint/issues/20

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -108,24 +107,152 @@ def test_get_zuul_yaml_files_skip():
...
+    jobs = zuulcilint_utils.get_zuul_object_from_yaml(ZuulObject.JOB, tmp_path / "job.yaml")
     assert len(jobs) == 1
     assert jobs[0]["job"]["name"] == "test-job"
 
 
-def test_get_jobs_from_zuul_yaml_invalid_yaml():
+def test_get_zuul_object_from_yaml_nodeset():
+    """Test that get_zuul_object_from_yaml() returns a list of nodesets."""
+    tmp_path = setup_zuul_job_yaml()
+    with pathlib.Path.open(tmp_path / "job.yaml", "w", encoding="utf-8") as f:
```
Comment (martimmatias) [inline tests/test_utils.py]: Why are you writing to a file named "job.yaml"? Shouldn't it be called "nodeset.yaml", for example?

Comment (pmpbaptista) [reply tests/test_utils.py]: Updated setup to dynamically create files and objects

File: tests/test_utils.py [hunk 2]
```diff
@@ -24,17 +25,19 @@ def setup_tmp_list_of_files():
...
         A Path object representing the temporary directory.
     """
     tmp_path = pathlib.Path(tempfile.mkdtemp())
-    with pathlib.Path.open(tmp_path / "job.yaml", "w", encoding="utf-8") as f:
+    with pathlib.Path.open(tmp_path / f"{obj_type.value}.yaml", "w", encoding="utf-8") as f:
         f.write(
-            """
+            f"""
+            - {obj_type.value}:
+                name: test-{obj_type.value}
```
Comment (martimmatias) [inline tests/test_utils.py:40]: The pragma object doesn't support the name field.

**Review by martimmatias — CHANGES_REQUESTED** [review]

File: tests/test_utils.py (hunk 2 above)
Comment (pmpbaptista) [reply tests/test_utils.py:40]: We're not testing the object itself, we're testing the ability to get the objects if they exist. To keep this tests generic, I guess we can keep this, if we need to test the object in the future, we can write directly to a new file

**Review by martimmatias — APPROVED** [review]

**Pull request merged**